engine/src/main/java/com/datatorrent/stram/client/StramClientUtils.java

```diff
@@ -739,13 +745,21 @@ public static InetSocketAddress getRMWebAddress(Configuration conf, String rmId)
 
   public static InetSocketAddress getRMWebAddress(Configuration conf, boolean sslEnabled, String rmId)
   {
-    rmId = (rmId == null) ? "" : ("." + rmId);
+    boolean isHA = (rmId != null);
+    rmId = isHA ? ("." + rmId) : "";
+    if (isHA) {
+      conf = getYarnConfiguration(conf);
+      conf.set(ConfigUtils.RM_HA_ID, rmId);
+    }
     InetSocketAddress address;
     if (sslEnabled) {
       address = conf.getSocketAddr(YarnConfiguration.RM_WEBAPP_HTTPS_ADDRESS + rmId, YarnConfiguration.DEFAULT_RM_WEBAPP_HTTPS_ADDRESS, YarnConfiguration.DEFAULT_RM_WEBAPP_HTTPS_PORT);
     } else {
       address = conf.getSocketAddr(YarnConfiguration.RM_WEBAPP_ADDRESS + rmId, YarnConfiguration.DEFAULT_RM_WEBAPP_ADDRESS, YarnConfiguration.DEFAULT_RM_WEBAPP_PORT);
     }
+    if (isHA) {
+      conf.unset(ConfigUtils.RM_HA_ID);
+    }
     LOG.info("rm webapp address setting {}", address);
```

Comment:
the conf.set and conf.unset combo may not have the expected behavior if getRMWebAddress() is called by multiple threads at the same time.